sys/xtimer/xtimer_core.c

```diff
@@ -142,8 +142,9 @@ void _xtimer_set(xtimer_t *timer, uint32_t offset)
         _shoot(timer);
     }
     else {
-        uint32_t target = _xtimer_now() + offset;
-        _xtimer_set_absolute(timer, target);
+        uint32_t now = _xtimer_now();
+        uint32_t target = now + offset;
+        _xtimer_set_absolute(timer, target, now);
```

Comment:
what if an interrupt or context switch happens between `_xtimer_now()` and `_xtimer_set_absolute`, IMHO this would mess up calculations in latter function or not?